packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -59,6 +60,10 @@ class FlutterCommandRunner extends CommandRunner<Null> {
     argParser.addFlag('version',
         negatable: false,
         help: 'Reports the version of this tool.');
+    argParser.addFlag('version-json',
```

Comment:
I think the user API would be a little cleaner if it was `flutter --version --json`.  You could add a check that enforces that `--json` is only supported if `--version` is specified.